File: 2024/july/java/03.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    private static int sum = 0;
    private ListNode solve(ListNode head){
        if (head == null) return null;

        ListNode cur = solve(head.next);
        sum+= head.val;
        if (head.val == 0 && head.next != null){
            ListNode ret = new ListNode(sum);
            ret.next = cur;
            sum = 0;
            return ret;
        }
        return cur;
    }
    public ListNode mergeNodes(ListNode head) {

        return solve(head);
    }
}

File: 2024/july/javaScript/03.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var mergeNodes = function(head) {
    let  sum = 0;
    function solve(head){
        if (head == null) return null;

        let cur = solve(head.next);
        sum+= head.val;
        if (head.val == 0 && head.next != null){
            let ret = new ListNode(sum);
            ret.next = cur;
            sum = 0;
            return ret;
        }
        return cur;
    }
    return solve(head);
    
};
